### PR TITLE
TST/BF: Skip inode tests if we could not change the inode.

### PR DESCRIPTION
**Pull request by bpoldrack**
Test for `AnnexRepo` with changed inode should not fail on NFS anymore.

This simply skips those tests, if the test's setup to achieve that inode change didn't work.

### CODE REVIEW

File: datalad/support/tests/test_gitrepo.py [hunk 1]
```diff
@@ -1105,7 +1105,10 @@ def test_GitRepo_flyweight_monitoring_inode(path, store):
     # recreate
     shutil.copytree(store, path, symlinks=True)
     new_inode = os.stat(path).st_ino
-    assert_not_equal(old_inode, new_inode)
+
+    if old_inode == new_inode:
+        raise SkipTest("inode did not change. Nothing to test for.")
```
Comment (yarikoptic) [inline datalad/support/tests/test_gitrepo.py]: oh perils of duplication

**Comment by yarikoptic** [comment]
some recent change brought a new ramification of the locking issue on python3... heh

**Comment by bpoldrack** [comment]
This is weird. Now it passes that test on NFS, too, but does NOT raise the `SkipTest`.

**Comment by bpoldrack** [comment]
Okay, lock-issue happened again, everything else seems fine. Thus merging.

**Pull request merged**